packages/mobile/src/app/saga.test.ts

```diff
@@ -71,6 +71,24 @@ describe('App saga', () => {
     expect(navigate).toHaveBeenCalledWith(Screens.BidaliScreen, { currency: CURRENCY_ENUM.DOLLAR })
   })
 
+  it('Handles openScreen deep link with safe origin', async () => {
+    const deepLink = `celo://wallet/openScreen?screen=${Screens.FiatExchangeOptions}&isAddFunds=true`
+    await expectSaga(handleDeepLink, openDeepLink(deepLink, true)).run()
+    expect(navigate).toHaveBeenCalledWith(
+      Screens.FiatExchangeOptions,
+      expect.objectContaining({ isAddFunds: true })
+    )
+  })
+
+  it('Handles openScreen deep link without safe origin', async () => {
+    const deepLink = `celo://wallet/openScreen?screen=${Screens.FiatExchangeOptions}&isAddFunds=true`
+    await expectSaga(handleDeepLink, openDeepLink(deepLink, false)).run()
+    expect(navigate).not.toHaveBeenCalledWith(
+      Screens.FiatExchangeOptions,
+      expect.objectContaining({ isAddFunds: true })
+    )
```

Comment:
To be safe, we should check no invocation occurred at all. 

For instance this successfully passes as a test:
```tsx
navigate(Screens.WebViewScreen, { uri: 'something' })
expect(navigate).not.toHaveBeenCalledWith(Screens.WebViewScreen, { uri: 'something else' })
```

```suggestion
    expect(navigate).not.toHaveBeenCalled()
```